### PR TITLE
[Formerly #487] Flag the use of the obsolete common type versions

### PR DESCRIPTION
**Pull request by bdefoy**
Formerly #487

### CODE REVIEW

File: docs/rules.md [hunk 1]
```diff
@@ -498,12 +482,22 @@ Location header must be supported for all async operations that return 202.
...
+
+Please refer to [lro-patch-provisioning-state-specified.md](./lro-patch-provisioning-state-specified.md) for details.
+
 ### LroPatch202
 
 Async PATCH should return 202.
 
 Please refer to [lro-patch202.md](./lro-patch202.md) for details.
 
+For long running (LRO) post operations, 'long-running-operation-options' must be present and have the 'final-state-via' property set to 'azure-async-operation'.
```
Comment (rkmanda) [inline docs/rules.md:497]: >final-state-via [](http://example.com/codeflow?start=103&length=15)

lets remove all rules related to final-state-via. Earlier there was a misunderstanding about what the final-state-via property was meant for. We no longer need this enforcement.

File: packages/rulesets/src/spectral/az-arm.ts [hunk 2]
```diff
@@ -684,6 +685,17 @@ const ruleset: any = {
     /// ARM rules without an RPC code
     ///
 
+    LatestVersionOfCommonTypesMustBeUsed: {
```
Comment (rkmanda) [inline packages/rulesets/src/spectral/az-arm.ts:688]: >LatestVersionOfCommonTypesMustBeUsed [](http://example.com/codeflow?start=4&length=36)

When this gets exercised in staging, lets verify that the rule gets enforced only for the new changes being put out and not for all references

**Review by rkmanda — CHANGES_REQUESTED** [review]
:clock1:

File: docs/rules.md (hunk 1 above)
Comment (bdefoy) [reply docs/rules.md:497]: Made an ADO item for this.

File: packages/rulesets/src/spectral/az-arm.ts (hunk 2 above)
Comment (bdefoy) [reply packages/rulesets/src/spectral/az-arm.ts:688]: Made an ADO item for this.

File: packages/rulesets/src/spectral/az-arm.ts [hunk 3]
```diff
@@ -699,6 +700,17 @@ const ruleset: any = {
...
+      description: "This rule checks for references that aren't using latest version of common-types.",
+      message: "{{error}}",
+      severity: "error",
```
Comment (bdefoy) [inline packages/rulesets/src/spectral/az-arm.ts:706]: set `stagingOnly`

Comment (rkmanda) [inline packages/rulesets/src/spectral/az-arm.ts:706]: >error [](http://example.com/codeflow?start=17&length=5)

change to warning

**Review by rkmanda — APPROVED** [review]
:shipit:

**Pull request merged**